sql/core/src/main/scala/org/apache/spark/sql/internal/SessionState.scala

```diff
@@ -79,7 +79,8 @@ private[sql] class SessionState(
     createQueryExecution: (LogicalPlan, CommandExecutionMode.Value) => QueryExecution,
     createClone: (SparkSession, SessionState) => SessionState,
     val columnarRules: Seq[ColumnarRule],
-    val queryStagePrepRules: Seq[Rule[SparkPlan]]) {
+    val queryStagePrepRules: Seq[Rule[SparkPlan]],
+    val runtimeOptimizerRules: Seq[Rule[LogicalPlan]]) {
```

Comment:
shall we have a wrapper for these custom AQE optimizations? otherwise we may keep adding new parameters here

Reply:
I agree it's worth. We can collect `queryStagePrepRules` and `runtimeOptimizerRules` into a new wrapper